cli/upload.py

```diff
@@ -202,19 +206,28 @@ def _wait_for_upload(procs: list, total: int) -> Optional[str]:
             # read stderr for this process
             errline = p.stderr.readline()
 
-            if p.poll() != None:
+            if p.poll() is not None:
                 p.stderr.close()
                 finished.add(i)
 
                 if p.returncode != 0:
-                    message += click.style(
-                        f"!!! upload error !!! ", fg="red", bold=True
-                    )
+                    message += click.style("!!! upload error !!! ", fg="red", bold=True)
                     message += p.args[-2]
                     click.echo(message)
+
+                    # stopping all other processes
+                    for other_p in procs:
+                        if p != other_p:
+                            other_p.kill()
+
+                    click.echo(
+                        f"\nGCS upload failed on {p.args[-2]} with the following message:\n"
+                    )
                     # Reconstruct multiline GCS error message
-                    error = f"{prev_errlines.get(i, '')}{errline}"
-                    break
+                    prev_err = prev_errlines.get(i, "")
+                    click.secho(f"{prev_err}{errline}", fg="red")
+
+                    raise click.Abort()
```

Comment:
Is this refactor fixing a bug, or is it just a code organization change? If the latter, I'm not sure it's worth the risk of inadvertently changing behavior.

Reply:
nope, just didn't revert far enough

Reply:
in next commit